project/buildSrc/src/main/kotlin/db.kt

```diff
@@ -264,6 +265,43 @@ open class PgAddData : AbstractTask() {
     }
 }
 
+open class PgInsertMasterToken : AbstractTask() {
+    @TaskAction
+    fun run() {
+        val randomString = TokenGenerator.generateRandomString()
+        val base64Token = TokenGenerator.encodeBase64(randomString)
+        val tokenHash = TokenGenerator.getHash(randomString)
+
+        val currTime = System.currentTimeMillis()
+        val expirationTime = currTime + 1000*60*60
+        val claims = "{\"client_id\":500, \"scope\": \"urn:org:ionproject:scopes:api:read\"}"
+
+        val insertQuery = """
+            INSERT INTO dbo.Token(hash,isValid,issuedAt,expiresAt,claims) VALUES ('$tokenHash',true,$currTime,$expirationTime,'$claims')
```

Comment:
I realize user input isn't used as parameters for this statement, but it still kind of bothers me that this is a possible SQL injection scenario. Maybe we should consider using the data source here?

Reply:
> Overall it looks good, but I have a few other questions:
> 
> * in the task `pgInsertMasterToken`, why are we specifically inserting a read scoped token? Couldn't it be of any other scope of choice?
> * why is the task named `pgInsertMasterToken`? The inserted token is of read scope and is similar to the insert.sql tokens, so why `master`?
> * the real database will not be reset everytime we deploy a new instance of the server. That said, is it OK for the `BootTokenGenerator` to insert a new `issue` token each and every time? (the previously generated `issue` scoped tokens will be left on the database as the server terminates)

1st Question:
Because there are only read operations, it would be pointless to insert a write token (see next answer)

2nd Question:
When I attributed the name I was thinking at long term, it will be indeed used to insert a master token but as I previously mentioned there are only read operations for now and more, the policies system currently in use will definitely change and im not sure how that will impact the application, so I opted to play with the safe side and let the master token be only the scope that already can do all operations.

3rd Question:
I realize the token will remain, thats why I have defined a function "cleanup..." lower in the code but not yet implemented.